### ISSUE TITLE
dataclass and swagger docs camelCase and snake_case transformations

### ISSUE THREAD

**Issue by romikforest** [Neoteroi/BlackSheep#173]
Hi.

If I declare some input dataclass using underscore names like:

```
@dataclass
class MyInput:
    ner_label: str
```

and I use it with `FromJSON` or directly as type annotation for some route function, I should pass payload using `ner_label` typing and not `nerLabel` otherwise I have a bad request exception:

`invalid parameter in request payload, caused by type MyInput or one of its subproperties. Error: got an unexpected parameter 'nerLabel'`

```
@post('/some_endpoint')
 async def some_endpoint(self, body: FromJSON[MyInput]) -> Response:
    body = body.value
     ...
```

```
@post('/some_endpoint')
 async def some_endpoint(self, body: MyInput) -> Response:
     ...
```

But if I generate docs using `EndpointDocs` and use `MyInput(ner_label='some_label')` in examples the `ner_label` is transformed in swagger to `nerLabel` and **Try it** functionality raises the error above.

Also if I declare dataclass using camelCase like:

```
@dataclass
class MyInput:
    nerLabel: str
```

The `nerLabel` is transformed in swagger to `nerlabel`

MacOS 11.3.1
Python 3.9.4
blacksheep-1.0.6-cp39-cp39-macosx_10_14_x86_64.whl

**Comment by RobertoPrevato** [comment]
Hi,
Can you please try to upgrade to the latest version of BlackSheep (1.0.7)?

Because I just tried to reproduce the error and I couldn't: in the screenshot below you can see a dataclass having a `ner_label` property is documented properly and gets the right value.

![image](https://user-images.githubusercontent.com/2576032/122275027-86d5f900-cee3-11eb-87ee-5bc10c5090d4.png)

In `1.0.7` there are many improvements for OpenAPI Documentation (among them, support for generating documentation for Generics and improved support for pydantic models), and @myusko added a great `TestClient` class that simplifies testing.

About the fact that automatic binding might handle the different naming conventions between JavaScript camelCase and Python snake_case, this is something I took into consideration but didn't have time to look into. I don't even know if `pydantic` supports this.

**Comment by myusko** [comment]
I think that pydantic can support this, but requires additional configuration.

e.g
```python
from pydantic import BaseModel
from humps import camelize


def to_camel(string):
    return camelize(string)


class Test(BaseModel):
    camel_case: str

    class Config:
        alias_generator = to_camel
        allow_population_by_field_name = True

Test(camel_case="here").dict(by_alias=True) # {'camelCase': 'here'}
```

**Comment by romikforest** [comment]
Hi.

I have upgraded to 1.0.7. and I my code is like this:

<img width="303" alt="изображение" src="https://user-images.githubusercontent.com/14263879/122291073-150aa900-cefd-11eb-80e0-90d2196b9462.png">

<img width="748" alt="изображение" src="https://user-images.githubusercontent.com/14263879/122291185-34a1d180-cefd-11eb-9d84-1a985245c6a9.png">

<img width="841" alt="изображение" src="https://user-images.githubusercontent.com/14263879/122291244-48e5ce80-cefd-11eb-9dee-9b386fcc5af0.png">

After that the endpoint waits for `ner_label`, but the generated swagger example is:

<img width="508" alt="изображение" src="https://user-images.githubusercontent.com/14263879/122291544-9cf0b300-cefd-11eb-907a-84a22c02f98c.png">

The **Try it** execution ends with an error:

<img width="1421" alt="изображение" src="https://user-images.githubusercontent.com/14263879/122291700-c7db0700-cefd-11eb-89c0-39a1f3f33b58.png">

**Comment by RobertoPrevato** [comment]
@romikforest thank you for providing an example to reproduce the issue, I see it now. The bug is in [essentials-openapi](https://github.com/Neoteroi/essentials-openapi) package. I will fix it as soon as possible and release a fix for both libraries.

**Comment by RobertoPrevato** [comment]
Hi @romikforest,
I worked to fix this bug and just published `blacksheep` 1.0.8 which references an updated version of `essentials-openapi`. Please upgrade to the latest version of the framework, things should be working now.

Contextually, I improved other elements, documented in the CHANGELOG.

Thanks again for reporting this bug!

PS. in the future I plan to add also built-in support for pydantic models when handling examples for OAD (they are currently supported by automatic binding but not for OAD generation).
